### DISCUSSION TITLE
Cost to support the ability for 3rd parties to download MRFs

### DISCUSSION THREAD

**Discussion by klomurray** [CMSgov/price-transparency-guide#170] [General]
Has anyone done calculations on the costs of these files being downloaded?

Given the file sizes and sheer number of files we can expect, I'm working to get a sense of the cost plans' would incur through the download / egress of these files. I’ve seen analysis for the cost to store the files but am concerned about the egress costs as these files are downloaded by 3rd parties. I’ve attached a sample calculator where you can play around with variables, but just as an example, if the insurer has 1,000 plans and based on estimates for egress costs ($10 per TB), file sizes, etc. it would cost $3,100 per month if a single entity were to download all the files each month ($37,200 per year). If we have a bunch of groups all trying to collect MRFs then these numbers add up really quick. 

A lot of this is based on the number of files given the plan_id requirements and the size of these files. If we had a schema that reduced the number of files by collecting plan ids together and made the files smaller similar to https://github.com/CMSgov/price-transparency-guide/discussions/99 then I don’t think this would be a problem but given the current lack of changes has anyone else done analysis on file download costs?

[MRF Download Costs.xlsx](https://github.com/CMSgov/price-transparency-guide/files/6978504/MRF.Download.Costs.xlsx)

**Comment by jeffrey-price** [comment]
@klomurray, I'd raised a similar concern in #147, [here](https://github.com/CMSgov/price-transparency-guide/discussions/147#discussioncomment-1023937). I didn't tease it out fully, but I was using the highest tier (best unit price - assuming that would be easily achieved) of AWS S3 Data Transfer Out ($0.05 / GB - call it $50 / tb).

Check me, but using other assumptions from that thread (150 plans x 3 files @ 1 TB ea x $50 / tb x 12 months @ 1 download / month = $270k / year). Many organizations will have significantly higher number of plans to support, considerably larger files, and download/download frequency that must be largely unrestricted to an unknown/undefined audience. I'm sure lower cost storage options exist (as in your example), but egress fees could easily eclipse the storage costs and taken together represent an enormous burden.

**Comment by ghost** [comment]
@klomurray Yes, I proposed a while back to reformat the schema in the posting Reformat Root Schema To Support Multiple Plan/HIOS/EIN and Less Individual Files #93 with no action by CMS.

What also needs to be taken into account since the files are publicly available and made available to the public without restrictions to download, you can have "bad actors" that could run scripts against payors/insurance carriers/self-funded groups that could financially cripple any organization that is not paying a capped monthly egress cost. Even with a capped monthly egress cost, the hosting organization bandwidth could be brought to a crawl.

**Reply by retjami** [reply]
Hi Bob, any updates on how you plan to handle bad actors? We are also concerned about excessive egress costs.

**Reply by ghost** [reply]
@retjami We are still evaluating what capabilities we have for this based on the ruling. The ruling should provide further definition on this taking into account the file producer and those that would host the file.
